### PR TITLE
NetworkCurl now uses RequestId in RequestHandle

### PR DESCRIPTION
**Pull request by meravingen123**
Resolves: OLPEDGE-1077

Signed-off-by: Serhii Lozynskyi <ext-serhii.lozynskyi@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/src/http/curl/NetworkCurl.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 #pragma once
 
+#include <curl/curl.h>
```
Comment (andescu) [inline olp-cpp-sdk-core/src/http/curl/NetworkCurl.h]: is this needed?

Comment (borzun) [reply olp-cpp-sdk-core/src/http/curl/NetworkCurl.h]: I guess yes, cause we use CURL related types in `NeworkCurl::RequestHandle` struct and in other callbacks.

Comment (andescu) [reply olp-cpp-sdk-core/src/http/curl/NetworkCurl.h]: But <curl/curl.h> is part of libcurl and should not be related to RequestId, the only change i see here.

Comment (andescu) [reply olp-cpp-sdk-core/src/http/curl/NetworkCurl.h]: Missed that it was just moved. Ignore my comment.

**Pull request merged**